components/sup/src/config.rs

```diff
@@ -39,6 +41,68 @@ pub const DEFAULT_ADDRESS_ENVVAR: &'static str = "HAB_LISTEN_GOSSIP";
 
 static LOGKEY: &'static str = "CFG";
 
+/// Enable the creation of a value based on an environment variable
+/// that can be supplied at runtime by the user.
+pub trait EnvConfig: Default + FromStr {
+    /// The environment variable that will be parsed to create an
+    /// instance of `Self`.
+    const ENVVAR: &'static str;
+
+    /// Generate an instance of `Self` from the value of the
+    /// environment variable `Self::ENVVAR`.
+    ///
+    /// If the environment variable is present and not empty, its
+    /// value will be parsed as `Self`. If it cannot be parsed, or the
+    /// environment variable is not present, the default value of the
+    /// type will be given instead.
+    fn configured_value() -> Self {
+        match henv::var(Self::ENVVAR) {
+            Err(VarError::NotPresent) => Self::default(),
+            Ok(val) => match val.parse() {
+                Ok(parsed) => {
+                    Self::log_parsable(&val);
+                    parsed
+                }
+                Err(_) => {
+                    Self::log_unparsable(&val);
+                    Self::default()
+                }
+            },
+            Err(VarError::NotUnicode(nu)) => {
+                Self::log_unparsable(nu.to_string_lossy());
+                Self::default()
+            }
+        }
+    }
+
+    /// Overridable function for logging when an environment variable
+    /// value was found and was successfully parsed as a `Self`.
+    ///
+    /// By default, we log a message at the `warn` level.
+    fn log_parsable(env_value: &String) {
+        warn!(
```

Comment:
My thinking was to have it be a more severe message so it would stand out if the variable actually was set. In that case, you'd definitely want to be aware that you're changing something.

Reply:
Good point. If the env var is not set, nothing is printed.